### PR TITLE
Add metrics exporter  

### PR DESCRIPTION
**Pull request by apentori**
Link to: 
- https://github.com/status-im/infra-role-postgres-ha/issues/12

Add 2 metrics exporter: 
* https://github.com/prometheus-community/postgres_exporter : for general Database metrics
* https://coroot.com/blog/engineering/pg-agent : for query performance metrics, disabled by  default to avoid impacting to much the database

### CODE REVIEW

File: tasks/consul.yml [hunk 1]
```diff
@@ -24,3 +24,25 @@
...
+        id:   '{{ postgres_ha_perf_metrics_exporter_cont_name }}'
+        tags: [ 'postgres-metrics', 'perf']
+        port: '{{ postgres_ha_perf_metrics_exporter_cont_port }}'
+        address: '{{ ansible_local.wireguard.vpn_ip }}'
+        checks:
+          - id: '{{ postgres_ha_perf_metrics_exporter_cont_name }}-status'
+            name: 'PostgreSQL Metrics exporter'
+            type: 'tcp'
+            tcp: 'localhost:{{ postgres_ha_perf_metrics_exporter_cont_port }}'
+            disabled: '{{ !postgres_ha_perf_metrics_exporter_enable | to_json }}'
```
Comment (jakubgs) [inline tasks/consul.yml]: We should probably just disable the whole service and not just the check if we just don't deploy it.

File: tasks/consul.yml [hunk 2]
```diff
@@ -24,3 +24,25 @@
             name:     'PostgreSQL {{ postgres_ha_replica_enabled | ternary("Replica", "Tables") }} Healthcheck'
             type:     'script'
             script:   '{{ postgres_ha_service_path }}/health.sh'
+      - name: 'postgres-metrics'
+        id:   '{{ postgres_ha_metrics_exporter_cont_name }}'
+        tags: [ 'postgres-metrics', 'perf']
+        port: '{{ postgres_ha_metrics_exporter_cont_port }}'
+        address: '{{ ansible_local.wireguard.vpn_ip }}'
```
Comment (jakubgs) [inline tasks/consul.yml]:
```suggestion
        address: '{{ ansible_local.wireguard.address }}'
```

File: tasks/consul.yml [hunk 3]
```diff
@@ -24,3 +24,25 @@
...
+        checks:
+          - id: '{{ postgres_ha_metrics_exporter_cont_name }}-status'
+            name: 'PostgreSQL Metrics exporter'
+            type: 'tcp'
+            tcp: 'localhost:{{ postgres_ha_metrics_exporter_cont_port }}'
+      - name: 'postgres-perf-metrics'
+        id:   '{{ postgres_ha_perf_metrics_exporter_cont_name }}'
+        tags: [ 'postgres-metrics', 'perf']
+        port: '{{ postgres_ha_perf_metrics_exporter_cont_port }}'
+        address: '{{ ansible_local.wireguard.vpn_ip }}'
```
Comment (jakubgs) [inline tasks/consul.yml]:
```suggestion
        address: '{{ ansible_local.wireguard.address }}'
```

**Review by jakubgs — APPROVED** [review]

File: defaults/main.yml [hunk 4]
```diff
@@ -51,6 +51,18 @@ postgres_ha_users: []
...
 #    pass: 'dbpass'
 
+# Metrics Exporter
+postgres_ha_metrics_exporter_cont_name: 'pg-metrics-exporter'
+postgres_ha_metrics_exporter_cont_version: 'v0.15.0'
+postgres_ha_metrics_exporter_cont_image: 'quay.io/prometheuscommunity/postgres-exporter:{{ postgres_ha_metrics_exporter_cont_version}}'
+postgres_ha_metrics_exporter_cont_port: 9187
+
+# Performence Metrics Exporter
+postgres_ha_perf_metrics_exporter_enable: False
```
Comment (yakimant) [inline defaults/main.yml]: `false` looks better

**Review by yakimant — APPROVED** [review]

**Pull request merged**